include/rocksdb/file_system.h

```diff
@@ -749,13 +749,29 @@ class FSWritableFile {
   //
   // PositionedAppend() requires aligned buffer to be passed in. The alignment
   // required is queried via GetRequiredBufferAlignment()
+
+  // Append data with verification information
+  virtual IOStatus AppendWithVerify(
```

Comment:
Please add a comment saying it is experimental and is subject to change.

We should think about a better name than `WithVerify()`. We probably can even still call it `Append()`.